app/src/androidTest/java/com/lightricks/feedexercise/data/ViewModelTest.kt

```diff
@@ -1,46 +1,100 @@
 package com.lightricks.feedexercise.data
 
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.lifecycle.testing.TestLifecycleOwner
 import androidx.room.Room
 import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.lightricks.feedexercise.data.Constans.hardCodedList
 import com.lightricks.feedexercise.database.FeedDatabase
-import com.lightricks.feedexercise.database.UserProject
 import com.lightricks.feedexercise.ui.feed.FeedViewModel
 import com.lightricks.feedexercise.ui.feed.FeedViewModelFactory
 import org.junit.Assert
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
-import java.util.concurrent.CountDownLatch
-import java.util.concurrent.TimeUnit
 
 @RunWith(AndroidJUnit4::class)
-class ViewModelTest (private var db : FeedDatabase,
-                     private var feedRepository: FeedRepository,
-                     private var userProjectList : List<UserProject>){
+class ViewModelTest {
+    private lateinit var db: FeedDatabase
+    private lateinit var constantfeedRepository: FeedRepository
+    private lateinit var increasingFeedRepo: FeedRepository
+    private val expectedFullLst = hardCodedList.map { item ->
+        FeedItem(item.id, responseUrlToThumbnailUrl(item.thumbnailUrl), item.isPremium)
+    }
+
+    private lateinit var viewModel: FeedViewModel
 
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
 
 
     @Before
-    fun init(){
+    fun init() {
         this.db = Room.inMemoryDatabaseBuilder(
             ApplicationProvider.getApplicationContext(),
-            FeedDatabase::class.java).build()
-        this.feedRepository = FeedRepository(MockApiService,db)
+            FeedDatabase::class.java
+        ).build()
+        this.constantfeedRepository = FeedRepository(ConstantItemsApiService, db)
+        this.increasingFeedRepo = FeedRepository(IncreasingItemsApiService, db)
+    }
+
+    @Test
+    fun testViewModelInitData() {
+        init()
+        val viewModelFactory = FeedViewModelFactory(constantfeedRepository)
+        viewModel = viewModelFactory.create(FeedViewModel::class.java)
+        Thread.sleep(100)
+        lateinit var viewModelLst: List<FeedItem>
+        viewModel.getFeedItems().observe(
+            TestLifecycleOwner(),
+            { feedItems -> viewModelLst = feedItems })
+        Assert.assertEquals(viewModelLst, expectedFullLst)
+    }
+
+
+    @Test
+    fun testViewModelDeleteData() {
+        init()
+        val viewModelFactory = FeedViewModelFactory(constantfeedRepository)
+        viewModel = viewModelFactory.create(FeedViewModel::class.java)
+        Thread.sleep(100)
+        deleteFirstProject()
```

Comment:
If this test wants to check that the VM reacts to a change in the data, then it should probably get feed items twice: before deleting and after.